tests/unit/s3/test_upload.py

```diff
@@ -172,31 +169,26 @@ async def test_upload_texts_with_metadata_http_error(self, upload_session_respon
                     "three.txt.meta.json",
                 ]
 
+        @pytest.mark.parametrize("status", [503, 502, 500, 504, 408])
+        @patch("aiohttp.ClientSession")
+        async def test_upload_file_retries_for_exception(
+            self, mock_session: Mock, upload_session_response: UploadSession, status: int
+        ) -> None:
+            exception = aiohttp.ClientResponseError(request_info=Mock(), history=Mock, status=status)  # type: ignore
+            with patch.object(aiohttp.ClientSession, "post", side_effect=exception):  # type: ignore
+                s3 = S3()
+
+                with pytest.raises(RetryableHttpError):
+                    await s3._upload_file_with_retries("one.txt", upload_session_response, "123", mock_session)
```

Comment:
Technically, since this is a private method, we should test the parent method that is calling this method or make it public. Not a blocker, but would be nice

Reply:
I don't think we should make the method public, but I also think its ok to test private methods in some cases